core/chains/evm/gas/docs/FEE_HISTORY_ESTIMATOR.md

```diff
@@ -0,0 +1,56 @@
+
+### Validations
+During startup, the estimator will perform two config checks:
+- `BumpPercent` is equal to or higher than *MinimumBumpPercentage*. *MinimumBumpPercentage* is fixed at 10% and it's the minimum percentage allowed by Geth when bumping a transaction, to prevent spam attacks. Replacing a transaction with a price less than 10% from the previous one will result in an error on the RPC side. Even for chains that don't enforce that rule, a 10% bump seems reasonable.
+- `RewardPercentile` is equal or lower than *ConnectivityPercentile*, when `EIP1559` is enabled. *ConnectivityPercentile* is fixed at the 85th percentile and it's the maximum percentile we're willing to bump a transaction's price. This is used as a sanity check method in order to avoid excessive gas bumping when an RPC is not responding.
+
+## As a Service
+`Fee History` estimator is built as a service. This means it will periodically poll the RPC for new prices, perform off-chain calculations, and cache the result for future use. For simplicity, only one type of gas estimation can be enabled at a time, Legacy or Dynamic. The poll interval is controlled by `CacheTimeout`. This value should be close to the block time. For slower chains, like Ethereum, you can set it to 12s, the same as the block time. For faster chains, you can skip a block or two, as prices will be refreshed more frequently. Ideally, 1s should be the absolute minimum. 
+
+
+## Legacy Gas Price Estimations
+### Fetching
+Periodically, `Fee History` estimator will call `eth_gasPrice` RPC method to fetch the gas price reported by the RPC. The parameters of this call can not be controlled by the user, meaning the result can sometimes be stale, especially during sudden gas spikes. It is advisable to use EIP-1559 if the chain supports it.
+
+### Bumping
+During bumping, `Fee History` will refresh the cached value by making a call to the RPC. The bumped value of the original price will be compared with the market price and the highest value will be returned.
+
+## Dynamic Price Estimations
+### Fetching
+`Fee History` estimator periodically calls `eth_feeHistory` method to get the most up-to-date information from the RPC (more information about the call can be found [here](https://ethereum.github.io/execution-apis/api-documentation/)). It fetches three things:
+- Base Fee of the next block
+- The Yth priority fee percentiles of the past X blocks, where Y is controlled by `RewardPercentile` and X by `BlockHistorySize`.
+- The 85th priority fee percentiles of the past X blocks.
+
+The above values are used to construct and cache the following:
+- **MaxPriorityFeePerGas**: the average of Yth priority fee percentiles, excluding zero values.
+- **MaxFeePerGas**: `baseFee * BaseFeeBufferPercentage + MaxPriorityFeePerGas`. BaseFeeBufferPercentage is used as a safety to catch any fluctuations in the Base Fee during the next blocks.
```

Comment:
nit: Since this is referring to a variable name, should we mark it with `
```suggestion
- **MaxFeePerGas**: `baseFee * BaseFeeBufferPercentage + MaxPriorityFeePerGas`. `BaseFeeBufferPercentage` is used as a safety to catch any fluctuations in the Base Fee during the next blocks.
```

Reply:
I used italics for const variables above, so I'll use the same here.